blog/_posts/2021-01-19-scala-native-0.4-release.md

```diff
@@ -0,0 +1,225 @@
+def fn(n: CInt): Unit = ???
+
+registerCallback(CFuncPtr1.fromScalaFunction(fn))
+registerCallback(fn)
+
+registerCallback { (n: CInt) => println("hello native") }
+```
+
+It is now also possible to work with an arbitrary pointer and convert it to `CFuncPtrN` that can be called in your Scala code, 
+or to convert your function to any pointer if your native library needs this.
+
+```scala
+import scala.scalanative.unsafe.Ptr
+val cFnPtr: CFuncPtr0[CInt]    = ???
+val fnPtr: Ptr[Byte]           = CFuncPtr.toPtr(cFnPtr)
+val fnFromPtr: CFuncPtr0[CInt] = CFuncPtr.fromPtr[CFuncPtr0[CInt]](fnPtr)
+```
+
+## New garbage collector
+This release also adds a new garbage collector - Commix, a parallel mark and concurrent sweep GC, based on the well known Immix GC.
+It reduces GC pause times by utilizing additional processor cores during mark and sweep phases. It was added soon after
+the 0.4.0-M2 release, but was only available via an unofficial snapshot.
+
+While the GC itself will use multiple threads, Scala Native still does not support multi-threading in the application code.
+Commix GC was written in C and uses `pthread` to work. In case your application needs concurrency support, you may try the experimental library [scala-native-loop](https://github.com/scala-native/scala-native-loop)
+
+## Sbt plugin changes
+### Introducing NativeConfig
+Upcoming Scala Native release would also bring changes to the sbt configuration. We've decided to make a sparse set of `NativeX` setting keys deprecated in favor of a single `NativeConfig`.
+
+```scala
+nativeConfig ~= (
+ _.withMode(build.Mode.releaseFast)
+  .withGC(build.GC.immix)
+  .withLTO(build.LTO.full)
+  .withOptimize(true)
+  .withCompileOptions(Nil)
+  .withLinkingOptions(Nil)
+ )
+```
+
+Old style settings keys are still supported, but with lower priority than the new config. They will be removed at some point in the future. In the following example resulting `LTO` setting would be set to `thin`
+
+```scala
+nativeConfig := nativeConfig.value
+ .withLTO(build.LTO.thin)
+
+nativeLTO := "none"
+```
+
+### Cross compiling
+By default, Scala Native has always compiled and optimized the resulting assembly based on host architecture and its operating system.
+Although it worked well when you were running a program on the same machine, it caused many problems when trying to run it anywhere else.
+With the latest release you're able to define a custom target for compiler by providing [LLVM style TargetTriple](https://clang.llvm.org/docs/CrossCompilation.html) in your config.
+
+For example, if you're working on Linux and would like to create an executable suitable for MacOS without changing your whole build,
+you can use the following sbt setting:
+```bash
+sbt 'set nativeConfig ~= {_.withTargetTriple("x86_64-apple-darwin<version>")}' myApp/nativeLink
+```
+
+We consider changing target triple as a feature for advanced users, and cannot promise it would currently work with any possible configuration yet.
```

Comment:
```suggestion
We consider changing the target triple as an experimental feature for advanced users, and cannot promise it will currently work with any possible configuration yet.
```